vesc_hw_interface/src/vesc_wheel_controller.cpp

```diff
@@ -70,13 +71,13 @@ void VescWheelController::control(const double target_velocity, const double cur
   }
 
   // overflow check
-  if (target_steps_ > static_cast<double>(LONG_MAX))
+  if (target_steps_ > std::numeric_limits<int>::max())
   {
-    target_steps_ += static_cast<double>(LONG_MIN);
+    target_steps_ += std::numeric_limits<int>::min();
   }
-  else if (target_steps_ < static_cast<double>(LONG_MIN))
+  else if (target_steps_ < std::numeric_limits<int>::min())
   {
-    target_steps_ += static_cast<double>(LONG_MAX);
+    target_steps_ += std::numeric_limits<int>::max();
   }
```

Comment:
The check will cause overflow `target_steps_` faster than `current_steps`.
i.e. `current_steps_ = std::numeric_limits<int>::max()` and `target_steps_ += ...`.

Could you check `target_steps_ - current_steps` over a threshold( such as `std::numeric_limits<int>::max`) and add `std::numeric_limits<int>::min - std::numeric_limits<int>::max` to `target_steps_`?

Reply:
Will this suffice?
```cpp
  if ((target_steps_ - current_steps) > std::numeric_limits<int>::max())
  {
    target_steps_ += std::numeric_limits<int>::min();
  }
  else if ((target_steps_ - current_steps) < std::numeric_limits<int>::min())
  {
    target_steps_ += std::numeric_limits<int>::max();
  }
```

Reply:
```cpp
  if ((target_steps_ - current_steps) > std::numeric_limits<int>::max())
  {
    target_steps_ += std::static_cast<double>(std::numeric_limits<int>::min()) - std::static_cast<double>(std::numeric_limits<int>::max());
	// or
	target_steps_ = std::numeric_limits<int>::min() + (target_steps - std::numeric_limits<int>::max());
  }
  else if ((target_steps_ - current_steps) < std::numeric_limits<int>::min())
  {
    target_steps_ += std::static_cast<double>(std::numeric_limits<int>::max()) - std::static_cast<double>(std::numeric_limits<int>::min());
	// or
	target_steps_ = std::numeric_limits<int>::max() + (target_steps - std::numeric_limits<int>::min());
  }
```